extensions/community/PokiGamesSDK.json

```diff
@@ -0,0 +1,321 @@
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "eventsFunctionContext.returnValue = gdjs._pokiGamesSdkExtension.isPlayingRewardedBreak || false;",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Checks if a commercial break just finished playing.",
+      "fullName": "Commercial break just finished playing",
+      "functionType": "Condition",
+      "group": "Poki Games",
+      "name": "DidCommercialBreakJustFinishPlaying",
+      "sentence": "Checks if a commercial break just finished playing",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "eventsFunctionContext.returnValue = gdjs._pokiGamesSdkExtension.commercialBreakJustFinishedPlaying;",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Checks if a rewarded break just finished playing.",
+      "fullName": "Rewarded break just finished playing",
+      "functionType": "Condition",
+      "group": "Poki Games",
+      "name": "DidRewardedBreakJustFinishPlaying",
+      "sentence": "Checks if a rewarded break just finished playing",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "eventsFunctionContext.returnValue = gdjs._pokiGamesSdkExtension.rewardedBreakJustFinishedPlaying;",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Checks if player should be rewarded after a rewarded break finished playing.",
+      "fullName": "Should reward player",
+      "functionType": "Condition",
+      "group": "Poki Games",
+      "name": "ShouldRewardPlayer",
+      "sentence": "Should reward player",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "eventsFunctionContext.returnValue = gdjs._pokiGamesSdkExtension.shouldRewardUser;",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "fullName": "",
+      "functionType": "Action",
+      "name": "onScenePostEvents",
+      "sentence": "",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "gdjs._pokiGamesSdkExtension.commercialBreakJustFinishedPlaying = false;",
+            "gdjs._pokiGamesSdkExtension.rewardedBreakJustFinishedPlaying = false;"
+          ],
```

Comment:
It could be done in PreEvents in case another extension wants to use the condition in its PostEvents.

Reply:
I'm not sure about that:
Both variables are being set to `true` when the promise returned by the poki SDK methods resolves. 4ian explained to me that those promises can only be resolved between 2 frames (because of the way JS works - single-threaded-ly), so between the post events of frame N and the pre events of frame N+1.
So these variables would be forever false when checked in the events.
Is it clear or is there something I'm missing?

Reply:
I see